src/main/java/com/github/black0nion/blackonionbot/commands/bot/HelpCommand.java

```diff
@@ -140,28 +139,28 @@ private void waitForHelpCatSelection(final @NotNull Message msg, final @NotNull
             } else {
                 final Category category = Category.valueOf(button.getId());
                 builder.setTitle(LanguageSystem.getTranslation("help", user, guild) + " | " + category.name().toUpperCase());
-                for (final Map.Entry<String[], TextCommand> entry : CommandBase.commandsArray.entrySet())
-                    if (entry.getValue().isVisible(user) && (entry.getValue().getCategory() == category))
-                        if (entry.getValue().getProgress() == Progress.DONE) {
-                            builder.addField(CommandEvent.getCommandHelp(guild, entry.getValue()), cmde.getTranslationOrEmpty("help" + entry.getValue().getCommand()[0]), false);
+                for (final Map.Entry<String, Pair<Long, SlashCommand>> entry : SlashCommandBase.commands.entrySet())
+                    if (entry.getValue().getValue().isHidden(user) && (entry.getValue().getValue().getCategory() == category))
```

Comment:
Please extract the `entry.getValue().getValue()` as it gets called a good few times here